### PR TITLE
STM32: Enable CAN bus on the stm32f3_disco development board

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add support for the CAN_1 CAN bus controller present in STM32F3 SoC series and enable it in the stm32f3_disco development board.

### CODE REVIEW

File: boards/arm/stm32f3_disco/stm32f3_disco.dts [hunk 1]
```diff
@@ -66,6 +66,7 @@
 		led0 = &green_led_6;
 		led1 = &green_led_7;
 		sw0 = &user_button;
+		can-1 = &can1;
```
Comment (alexanderwachter) [inline boards/arm/stm32f3_disco/stm32f3_disco.dts]: Can you change it to can-primary. #22141

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

File: boards/arm/stm32f3_disco/stm32f3_disco.dts (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts]: Yes, if #22141 is merged before this, I will. Otherwise the samples won't work.

Comment (alexanderwachter) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts]: The sample does not use the alias, but the node directly.

Comment (henrikbrixandersen) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts]: Some samples, yes. Others (like the canopen sample) uses the aliases.

**Review by alexanderwachter — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

File: boards/arm/stm32f3_disco/stm32f3_disco.dts (hunk 1 above)
Comment (alexanderwachter) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts]: @henrikbrixandersen #22141  is merged.

Comment (henrikbrixandersen) [reply boards/arm/stm32f3_disco/stm32f3_disco.dts]: @alexanderwachter Fixed, thanks.

**Review by erwango — APPROVED** [review]

**Pull request merged**